### ISSUE TITLE
Missing Aria-Label on the button to the right of the integration name. 

### ISSUE THREAD

**Issue by digitaldarragh** [home-assistant/frontend#16978]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Screen readers are unable to clearly identify the purpose of the button to the right of each integration. title on the integrations page. Each button is read as: "Unlabeled 0 button".  
Perhaps these buttons should be set to aria-hidden?  Especially if the button's purpose is already handled by clicking the name of the integration. 

### Describe the behavior you expected

Either the button should be hidden if it saves no unique purpose. Or the button should be properly labeled with the aria-label attribute. 

### Steps to reproduce the issue

1.
2.
3.
...
Using a screen reader such as Jaws, NVDA, Voiceover or talk back, you will notice this issue when navigating through the Integrations page. 

### What version of Home Assistant Core has the issue?

2023.7

### In which browser are you experiencing the issue with?

All

### Which operating system are you using to run this browser?

All
